proposals/017-shader-record.md

```diff
@@ -249,20 +284,29 @@ Proposed Approach
     `inout` being the default. 
     
     This would resolve the ambiguity regarding `in` and implicit `uniform` incorrectly mapping to hit attributes
-    and shader records. 
+    and shader records.  -->
 
 
 Detailed Explanation
 --------------------
-1. **Language-Level Model**
+<!-- 1. **Language-Level Model**
     In a future ideal version of Slang, we might introduce explicit syntax for 
     specifying data "rates," e.g. `[[thread_group]]`, `[[wave]]`, `[[lane]]`, or more. 
     This is consistent with advanced GPU programming models that differentiate per-thread, 
     per-wavefront, per-group, per-dispatch, etc. However, this proposal focuses on 
     disambiguating push constant uniforms from shader record uniforms, as these are the most 
-    pressing distinction.
-
-2. **Entry-Point Parameter Rules**
+    pressing distinction. -->
+
+1. **Entry-Point Parameter Rules**
+    * Parameters declared `uniform` at entry points always map to push-constant-like storage
+      (push constants, root constants, or a small constant buffer) regardless of stage/backend.
+    * Parameters typed as `ShaderRecord<T>` map to per-shader-record data (Vulkan/OptiX SBT;
+      D3D local root signature).
+    * Parameters typed as `Payload<T>` map to payload registers. `in`, `out`, and `inout` apply,
```

Comment:
My understanding of these wrapper types is that they should be treated as if they are pointer types. So `Payload<T>` is effectively a `T*`. It is not meaningful to have `inout T*` or `out T*` because that just means the pointer itself is mutable. Instead, we probably want to have `RWPayload<T>`, `WPayload<T>` to represent payload pointers with read/write or write-only access.

Reply:
We should also just make it invalid to use `in` `inout` or `out` on ray tracing entrypoint parameters.

Reply:
I’m not sure I’m aligned on the RWPayload though. With PAQs, users annotate individual fields within the structure passed as T as being read, write, or read write. It generally never makes sense to make the entire payload read-only. 

My thought was to instead look at the T being passed into the payload type and assume that that’s where PAQs information comes from. 

Then otherwise drop in/out semantics entirely